src/altair/helpers.rs

```diff
@@ -68,9 +103,36 @@ pub fn get_next_sync_committee<
         MAX_VALIDATORS_PER_COMMITTEE,
         SYNC_COMMITTEE_SIZE,
     >,
-    _context: &Context,
-) -> Result<()> {
-    Ok(())
+    context: &Context,
+) -> Result<SyncCommittee<SYNC_COMMITTEE_SIZE>> {
+    // Return the next sync committee, with possible pubkey duplicates.
+    let indices = get_next_sync_committee_indices(state, context)?;
+    let public_keys = state
+        .validators
+        .iter()
+        .enumerate()
+        .filter_map(|(i, v)| {
+            if indices.contains(&i) {
+                Some(&v.public_key)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+    // @dev not sure if the impl of `eth_aggregate_pubkeys` is needed or correct
+    // also, not sure if `expect` is desired here
```

Comment:
why not call `eth_aggregate_pubkeys`?

also I would use `?` operator rather than `expect` here

Reply:
Oops -- should have clarified. In `crypto`, I've added the impl for `eth_aggregate_pubkeys` (which I'll rename to `eth_aggregate_public_keys` for consistency purposes in this library). I'm not sure if _this_ is the correct impl. Looks like it is (potentially) correct via your comment https://github.com/ralexstokes/ethereum-consensus/pull/106/files#r888586787

Reply:
Re: `?` operator -- I get an error if I try to do this: `'?' couldn't convert the error to 'state_transition::error::Error'`

```
the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
the following implementations were found:
  <state_transition::error::Error as std::convert::From<ssz_rs::MerkleizationError>>
  <state_transition::error::Error as std::convert::From<ssz_rs::SimpleSerializeError>>
  <state_transition::error::Error as std::convert::From<std::boxed::Box<state_transition::error::InvalidBlock>>>
required because of the requirements on the impl of `std::ops::FromResidual<std::result::Result<std::convert::Infallible, crypto::Error>>` for `std::result::Result<altair::sync::SyncCommittee<SYNC_COMMITTEE_SIZE>, state_transition::error::Error>`rustc[E0277](https://doc.rust-lang.org/error-index.html#E0277)
helpers.rs(107, 6): expected `state_transition::error::Error` because of this
```

The only "new" thing `eth_aggregate_pubkeys` uses is `AggregatePublicKey::aggregate`, which has the same `BLSTError` as the other fns in `crypto`, so I don't think that's the issue. It also returns a Result with `PublicKey`, which I don't think any others in `crypto` return. Is that potentially the issue? E.g., `state_transition::error` needs something like `InvalidPublicKey(BlsPublicKey)`?

Otherwise, it looks like it's coming from `SyncCommittee<SYNC_COMMITTEE_SIZE>, state_transition::error::Error`?

Reply:
yeah I'm a little surprised this is not already there but we want to do something like this:

https://github.com/ralexstokes/ethereum-consensus/blob/main/src/state_transition/error.rs#L13-L14

so there is another enum variant `Crypto(#[from] CrytpoError)` if you do something like `use crate::crypto::Error as CryptoError;`

Reply:
ahh that was easy lol